### ISSUE TITLE
UnusedTypeParams - NoSuchMethodError: 'scala.meta.Type$Bounds scala.meta.Type$Param.bounds()'

### ISSUE THREAD

**Issue by SuperFola** [xuwei-k/scalafix-rules#259]
Using a `.scalafix.conf` with the following content:

```
rules = [
  UnusedTypeParams
]
```

and a Test.scala:

```scala
package test

object Test {
  def test[A, B](a: A, b: B, build: (A, B) => String): String = build(a, b)
}
```

After more testing it seems that it can be simplified even more to `def test[A](a: A): String = a.toString`. Moving the type `[A]` to a `class Test[A]` doesn't generate the error.

Using sbt 1.10.7, Scala 2.13.15, Azul Systems, Inc. Java 17.0.1 (on M1 Pro Mac).

It generates an error `NoSuchMethodError: 'scala.meta.Type$Bounds scala.meta.Type$Param.bounds()'` with the following stacktrace
<details>
<summary>Stacktrace</summary>
<pre><code>[error] java.lang.NoSuchMethodError: 'scala.meta.Type$Bounds scala.meta.Type$Param.bounds()'
[error] 	at fix.UnusedTypeParams$$anonfun$fix$1.$anonfun$applyOrElse$3(UnusedTypeParams.scala:26)
[error] 	at fix.UnusedTypeParams$$anonfun$fix$1.$anonfun$applyOrElse$3$adapted(UnusedTypeParams.scala:26)
[error] 	at scala.collection.immutable.List.filter(List.scala:516)
[error] 	at fix.UnusedTypeParams$$anonfun$fix$1.applyOrElse(UnusedTypeParams.scala:26)
[error] 	at fix.UnusedTypeParams$$anonfun$fix$1.applyOrElse(UnusedTypeParams.scala:21)
[error] 	at scala.PartialFunction$Lifted.apply(PartialFunction.scala:338)
[error] 	at scala.PartialFunction$Lifted.apply(PartialFunction.scala:334)
[error] 	at scala.meta.transversers.Api$XtensionCollectionLikeUI$traverser$3$.apply(Api.scala:30)
[error] 	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1(SimpleTraverser.scala:5)
[error] 	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1$adapted(SimpleTraverser.scala:5)
[error] 	at scala.collection.immutable.List.foreach(List.scala:334)
[error] 	at scala.meta.transversers.SimpleTraverser.apply(SimpleTraverser.scala:5)
[error] 	at scala.meta.transversers.Api$XtensionCollectionLikeUI$traverser$3$.apply(Api.scala:31)
[error] 	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1(SimpleTraverser.scala:5)
[error] 	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1$adapted(SimpleTraverser.scala:5)
[error] 	at scala.collection.immutable.List.foreach(List.scala:334)
[error] 	at scala.meta.transversers.SimpleTraverser.apply(SimpleTraverser.scala:5)
[error] 	at scala.meta.transversers.Api$XtensionCollectionLikeUI$traverser$3$.apply(Api.scala:31)
[error] 	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1(SimpleTraverser.scala:5)
[error] 	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1$adapted(SimpleTraverser.scala:5)
[error] 	at scala.collection.immutable.List.foreach(List.scala:334)
[error] 	at scala.meta.transversers.SimpleTraverser.apply(SimpleTraverser.scala:5)
[error] 	at scala.meta.transversers.Api$XtensionCollectionLikeUI$traverser$3$.apply(Api.scala:31)
[error] 	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1(SimpleTraverser.scala:5)
[error] 	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1$adapted(SimpleTraverser.scala:5)
[error] 	at scala.collection.immutable.List.foreach(List.scala:334)
[error] 	at scala.meta.transversers.SimpleTraverser.apply(SimpleTraverser.scala:5)
[error] 	at scala.meta.transversers.Api$XtensionCollectionLikeUI$traverser$3$.apply(Api.scala:31)
[error] 	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1(SimpleTraverser.scala:5)
[error] 	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1$adapted(SimpleTraverser.scala:5)
[error] 	at scala.collection.immutable.List.foreach(List.scala:334)
[error] 	at scala.meta.transversers.SimpleTraverser.apply(SimpleTraverser.scala:5)
[error] 	at scala.meta.transversers.Api$XtensionCollectionLikeUI$traverser$3$.apply(Api.scala:31)
[error] 	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1(SimpleTraverser.scala:5)
[error] 	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1$adapted(SimpleTraverser.scala:5)
[error] 	at scala.collection.immutable.List.foreach(List.scala:334)
[error] 	at scala.meta.transversers.SimpleTraverser.apply(SimpleTraverser.scala:5)
[error] 	at scala.meta.transversers.Api$XtensionCollectionLikeUI$traverser$3$.apply(Api.scala:31)
[error] 	at scala.meta.transversers.Api$XtensionCollectionLikeUI.collect(Api.scala:34)
[error] 	at fix.UnusedTypeParams.fix(UnusedTypeParams.scala:21)
[error] 	at scalafix.internal.v1.Rules.$anonfun$syntacticPatch$3(Rules.scala:91)
[error] 	at scala.collection.immutable.List.map(List.scala:247)
[error] 	at scalafix.internal.v1.Rules.syntacticPatch(Rules.scala:90)
[error] 	at scalafix.internal.v1.MainOps$.getPatchesAndDiags(MainOps.scala:315)
[error] 	at scalafix.internal.v1.MainOps$.unsafeHandleFile(MainOps.scala:256)
[error] 	at scalafix.internal.v1.MainOps$.handleFile(MainOps.scala:361)
[error] 	at scalafix.internal.v1.MainOps$.$anonfun$run$3(MainOps.scala:405)
[error] 	at scalafix.internal.v1.MainOps$.$anonfun$run$3$adapted(MainOps.scala:399)
[error] 	at scala.collection.immutable.Vector.foreach(Vector.scala:2125)
[error] 	at scalafix.internal.v1.MainOps$.run(MainOps.scala:399)
[error] 	at scalafix.internal.v1.MainOps$.run(MainOps.scala:79)
[error] 	at scalafix.internal.interfaces.ScalafixArgumentsImpl.run(ScalafixArgumentsImpl.scala:38)
[error] 	at scalafix.internal.sbt.ScalafixInterface.run(ScalafixInterface.scala:109)
[error] 	at scalafix.sbt.ScalafixPlugin$.$anonfun$runArgs$7(ScalafixPlugin.scala:709)
[error] 	at scalafix.sbt.ScalafixPlugin$.$anonfun$runArgs$7$adapted(ScalafixPlugin.scala:706)
[error] 	at scalafix.sbt.ScalafixPlugin$.accumulateAndRunForStaleTargets$1(ScalafixPlugin.scala:684)
[error] 	at scalafix.sbt.ScalafixPlugin$.$anonfun$runArgs$4(ScalafixPlugin.scala:691)
[error] 	at scalafix.sbt.ScalafixPlugin$.$anonfun$runArgs$3(ScalafixPlugin.scala:675)
[error] 	at sbt.util.Difference.apply(Tracked.scala:415)
[error] 	at sbt.util.Difference.apply(Tracked.scala:390)
[error] 	at scalafix.sbt.ScalafixPlugin$.diffTargets$1(ScalafixPlugin.scala:674)
[error] 	at scalafix.sbt.ScalafixPlugin$.$anonfun$runArgs$6(ScalafixPlugin.scala:699)
[error] 	at scalafix.sbt.ScalafixPlugin$.accumulateAndRunForStaleTargets$1(ScalafixPlugin.scala:686)
[error] 	at scalafix.sbt.ScalafixPlugin$.$anonfun$runArgs$1(ScalafixPlugin.scala:701)
[error] 	at scalafix.sbt.ScalafixPlugin$.$anonfun$runArgs$1$adapted(ScalafixPlugin.scala:663)
[error] 	at sbt.util.Tracked$.$anonfun$inputChangedW$1(Tracked.scala:220)
[error] 	at scalafix.sbt.ScalafixPlugin$.diffWithPreviousRuns$1(ScalafixPlugin.scala:703)
[error] 	at scalafix.sbt.ScalafixPlugin$.runArgs(ScalafixPlugin.scala:706)
[error] 	at scalafix.sbt.ScalafixPlugin$.$anonfun$scalafixSyntactic$1(ScalafixPlugin.scala:503)
[error] 	at scalafix.sbt.ScalafixPlugin$.$anonfun$scalafixSyntactic$1$adapted(ScalafixPlugin.scala:499)
[error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:63)
[error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:69)
[error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:283)
[error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:24)
[error] 	at sbt.Execute.work(Execute.scala:292)
[error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:283)
[error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:65)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
[error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
</code></pre>
</details>

It seems part of the rule is redundant too:

https://github.com/xuwei-k/scalafix-rules/blob/2e6ea2c8e2d16ee87e656590744b776ef39d82b9/rules/src/main/scala/fix/UnusedTypeParams.scala#L26

`p.bounds.context.isEmpty` is duplicated, once at the beginning and once at the end of the rule.
